lib/importers/twilio/saver.rb

```diff
@@ -5,15 +5,27 @@ def initialize(calls:)
         @calls = calls
       end
 
-      def store_valid_by_date
+      def save
+        ActiveRecord::Base.transaction do
+          save_calls!
+          save_valid_by_date!
+        end
+      end
+
+      def save_calls!
+        @calls.each do |call|
+          TwilioCall.find_by(uid: call.uid) ||
```

Comment:
Unfortunately not. The params related to location, etc can change over time. Using find_or_create_by would result in duplicate records being created (potentially).